### DISCUSSION TITLE
Add FreshRSS behind Authentik with OpenID Connect with LinuxServer.io image

### DISCUSSION THREAD

**Discussion by lachnerd** [FreshRSS/FreshRSS#6873] [Q&A] [answered]
### Describe the bug

Hi there, 
i wanted to handle freshrss behind auhtentik to use openid logon.

So i followed the documentation configured authentik accordingly, made sure my freshrss-user has admin  rights and is identical to authentic username. 

After switching to HTTP Auth:
<img width="478" alt="image" src="https://github.com/user-attachments/assets/ba2de5ec-2a93-408f-85a6-178eb222d50a">

I am locked out end onyl get:
"[HTTP Remote-User={{ .Request.Header.Get “Remote-User” }} ; Remote IP address=10.42.1.48]"

And i'm running freshrss authentik etc. in k8s accessed behind traefik.
I didn't have to configure any fancy middlewares or stuff for my other existing applications.

### To Reproduce

1. add freshrss to authentic
2. add ENV Variables to freshrss with key & secret
3. Restart & anable http auth
4. lock out

### Expected behavior

Authentication gets routed to auhtentik app

### FreshRSS version

1.24.3

### Environment information

k3s: 1.30
authentik: 2024.8.2
freshrss: 1.24.3

**Comment by Alkarex** [comment]
Hello,
If you see `Remote-User={{ .Request.Header.Get “Remote-User” }}` then the username received is most likely wrong. Maybe a curly bracket missing or something like that in your template.
It is expected to see a correct username there

**Comment by lachnerd** [comment]
> Hello, If you see `Remote-User={{ .Request.Header.Get “Remote-User” }}` then the username received is most likely wrong. Maybe a curly bracket missing or something like that in your template. It is expected to see a correct username there

thx, i will check again authentik, but as far as i know there is no template 

Is there an easy way to undo authentikation to webauth ? -  its pretty annoying to test right now

**Comment by AnonymousX86** [comment]
Not sure if I should create another issue, but I'm also trying to put FreshRSS behind Authentik. I've followed official docs [this](https://docs.goauthentik.io/integrations/services/freshrss/) then [this](https://freshrss.github.io/FreshRSS/en/admins/16_OpenID-Connect-Authentik.html), eventually only latter one, but it seems like `Remote-User` is completely missing.
![image](https://github.com/user-attachments/assets/a88f5d4b-7d86-4de3-9d62-89d344291cdd)
`127.18.0.1` is my Nginx proxy (LinuxServer.io SWAG).

EDIT: Sorry, I was using LinuxServer.io image, which (I assume) is based on Alpine.

**Comment by lachnerd** [comment]
> Not sure if I should create another issue, but I'm also trying to put FreshRSS behind Authentik. I've followed official docs [this](https://docs.goauthentik.io/integrations/services/freshrss/) then [this](https://freshrss.github.io/FreshRSS/en/admins/16_OpenID-Connect-Authentik.html), eventually only latter one, but it seems like `Remote-User` is completely missing. ![image](https://private-user-images.githubusercontent.com/53279236/374368412-a88f5d4b-7d86-4de3-9d62-89d344291cdd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzNzA3NTUsIm5iZiI6MTcyODM3MDQ1NSwicGF0aCI6Ii81MzI3OTIzNi8zNzQzNjg0MTItYTg4ZjVkNGItN2Q4Ni00ZGUzLTlkNjItODlkMzQ0MjkxY2RkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA4VDA2NTQxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyODU1MjE1YmE2NjNkZGZjNjcxZjY0MmUyOTFiYjYwYzFhMjQ4OTNkNjA0ZDg1ZWU2ZmEyNTkzZDJlZTUyOTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.BL3m16zrW8Xp8v10kKoj3mOwr7fEtuWXzTr2jxv4baY) `127.18.0.1` is my Nginx proxy (LinuxServer.io SWAG).
> 
> EDIT: Sorry, I was using LinuxServer.io image, which (I assume) is based on Alpine.

wait - i am also using LinuxServer.io image - this isn't supported here - then maybe thats my problem....

**Comment by lachnerd** [comment]
ok authentik docs is pretty clear on this...:
<img width="728" alt="image" src="https://github.com/user-attachments/assets/5b180f98-e76b-4efc-92ca-54f5106592de">

my bad, can be closed

**Comment by Alkarex** [comment, marked as answer]
Indeed, at the moment, only the official `freshrss/freshrss` images support OpenID Connect, and only in their default (Debian) variant, not the Alpine version (yet), and not for the ARM platform (yet).

See also https://freshrss.github.io/FreshRSS/en/admins/16_OpenID-Connect.html

I hope that these configurations will soon be possible. Help welcome to follow-up on the situation:
https://github.com/FreshRSS/FreshRSS/issues/5722

The support for ARM on Debian should be possible as soon as the package gets updated to [mod_auth_openidc 2.4.15+](https://github.com/OpenIDC/mod_auth_openidc/releases/tag/v2.4.15)
https://packages.debian.org/bookworm/libapache2-mod-auth-openidc

And support for Alpine (including ARM) as soon as the package gets promoted from `edge` to a stable version
https://pkgs.alpinelinux.org/packages?name=apache-mod-auth-openidc

Meanwhile for Alpine / ARM, our development Dockerfile [`latest`](https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/Dockerfile-Newest) can be experimented with

Feedback welcome on whether you have managed to make your setup work

**Reply by Alkarex** [reply]
Help welcome:
* https://github.com/FreshRSS/FreshRSS/issues/8031

**Comment by lachnerd** [comment]
my setup is working now with the standard freshrss/freshrss image, thx

**Reply by SnoopAir** [reply]
hey @lachnerd I'm still struggeling aka failing to make it to the finish line. I'm testing different combinations starting from scratch rather often but never succeed. Something is always off. Could you give me some hints how you eventually got it working?
My setup is also freshrss currently in version 1.25.0, traefik and authentik.

**Reply by lachnerd** [reply]
hey @SnoopAir, for me the key here was using non-alpine images and setting freshrss auth-method to "HTTP" - if you need further details just let me know...

**Reply by SnoopAir** [reply]
Thanks @lachnerd . Using the alpine image was also my first and failed attempt. I pretty quickly changed to the regular ones, but failed anyway. 
So my first question would be: Did you start the setup from the standard installation without using authentik/oidc integration and switched over to it in a 2nd step? Or did you start from scratch with env already setup with OIDC integration from the beginning?

**Reply by lachnerd** [reply]
sorry didn't see that last question....

as i had to re setup i realized the following:
- initial OIDC setup is working from beginning using HTTP Auth Option
- can also be later changed i guess but not via GUI afaik

And i realized best thing is to wait till freshrss has completely started up - i had to take several turns which ended up in 504s and  stuff like that only because i was to fast i guess.

the freshrss setup seems still not very rock solid - but if you got it  working its a cool tool
